### PR TITLE
SISRP-17106 - Adds EdoOracle::Queries#has_instructor_history?

### PR DESCRIPTION
**Pull request by redconfetti**
https://jira.berkeley.edu/browse/SISRP-17106

Implement EDO DB equivalent of CampusOracle::Queries#has_instructor_history?

### CODE REVIEW

File: spec/models/edo_oracle/queries_spec.rb [hunk 1]
```diff
@@ -118,4 +134,16 @@
     end
   end
 
+  describe '.has_instructor_history?', :textext => true do
+    subject { EdoOracle::Queries.has_instructor_history?(ldap_uid, terms) }
+    context 'when user is not an instructor' do
```
Comment (paulkerschen) [inline spec/models/edo_oracle/queries_spec.rb]: @redconfetti Is the 'not' in here a typo?

File: app/models/edo_oracle/connection.rb [hunk 2]
```diff
@@ -11,5 +11,11 @@ def self.settings
     def self.stringified_columns
       %w(section_id campus-uid)
     end
+
+    def self.terms_query_list(terms = nil)
+      terms.try :compact!
+      return '' unless terms.present?
+      terms.map { |term| "'#{term.campus_solutions_id}'" }.join ','
+    end
```
Comment (paulkerschen) [inline app/models/edo_oracle/connection.rb]: +1

File: app/models/edo_oracle/queries.rb [hunk 3]
```diff
@@ -285,5 +285,26 @@ def self.get_enrolled_students(section_id, term_id)
...
+          SISEDO.ASSIGNEDINSTRUCTORV00_VW instr
+        WHERE
+          instr."campus-uid" = '#{ldap_uid}' AND
+          rownum < 2 AND
+          instr."term-id" IN (#{terms_list})
+        SQL
+        result = connection.select_one(sql)
+      }
+      Rails.logger.debug "Instructor #{ldap_uid} history for terms #{instructor_terms} count = #{result}"
+      return result["course_count"].to_i > 0
```
Comment (paulkerschen) [inline app/models/edo_oracle/queries.rb]: @redconfetti don't need no `return` at method's end.

Comment (redconfetti) [inline app/models/edo_oracle/queries.rb]: oh yeah, that was a copy-paste from CampusOracle

File: spec/models/edo_oracle/queries_spec.rb (hunk 1 above)
Comment (redconfetti) [inline spec/models/edo_oracle/queries_spec.rb]: Yes. Typo

File: app/models/edo_oracle/connection.rb (hunk 2 above)
Comment (redconfetti) [inline app/models/edo_oracle/connection.rb]: @paulkerschen Yeah. I noticed that your line didn't guard against an empty `terms` argument

**Pull request merged**